redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts

```diff
@@ -280,4 +280,43 @@ export class RecommendationProvider {
       return null;
     }
   }
+
+  /**
+ * Check set password recommendation
+ * @param redisClient
+ */
+
+  async determineSetPasswordRecommendation(
+    redisClient: Redis | Cluster,
+  ): Promise<Recommendation> {
+    if (await this.checkAuth(redisClient)) {
+      return { name: RECOMMENDATION_NAMES.SET_PASSWORD };
+    }
+
+    try {
+      const users = await redisClient.sendCommand(
+        new Command('acl', ['list'], { replyEncoding: 'utf8' }),
+      ) as string[];
+
+      const nopassUser = users.some((user) => user.split(' ')[3] === 'nopass');
+
+      return nopassUser ? { name: RECOMMENDATION_NAMES.SET_PASSWORD } : null;
+    } catch (err) {
+      this.logger.error('Can not determine set password recommendation', err);
+      return null;
+    }
+  }
+
+  private async checkAuth(redisClient: Redis | Cluster): Promise<boolean> {
+    try {
+      await redisClient.sendCommand(
+        new Command('auth', ['pass']),
```

Comment:
what will happen if password is set to 'pass' in fact? (seems like nothing harmful, but...)
Here you just trying to catch if AUTH disabled at all?
Interesting thing here for which user you will send auth command? is here any user required?

Reply:
Here i trying to check is any password set. This command will send for current connected user. If in any reason this command will executed it will not change anything. Let me double check